ADR/0011-roles-and-permissions.md

```diff
@@ -57,6 +59,7 @@ We will use the built-in Kubernetes RBAC system for Stonesoup's role and permiss
 | Admin         | Workspace               | Access to namespaces that backs workspace                           |
 |               | Application & Component | appstudio.redhat.com      | *                                       | applications, components, componentdetectionqueries
 |               | Environment             | appstudio.redhat.com      | *                                       | promotionruns, snapshotenvironmentbindings, snapshots, environments
+|               | DeploymentTarget       | appstudio.redhat.com      | get, list, watch                        | deploymentttargets, deploymenttargetclaims, deploymenttargetclasses
```

Comment:
```suggestion
|               | DeploymentTarget        | appstudio.redhat.com      | *                        | deploymentttargets, deploymenttargetclaims
|               | DeploymentTargetClass   | appstudio.redhat.com      | get, list, watch                        | deploymenttargetclasses
```

For BYOC (bring your own compute/bring your own cluster), the Admin needs to be able to create an Environment, create a DT, and create a DTC. The admin user already has `*` on Environment. We should set DeploymentTarget and DeploymentTargetClaim to match.

However, the Admin should _not_ be able to create or delete deploymenttargetclasses, which (I think) are cluster scoped.

Reply:
two comments:
1. what you have written above doesn't match what is in the "admin" template now: https://github.com/codeready-toolchain/host-operator/blob/master/deploy/templates/nstemplatetiers/appstudio/spacerole_admin.yaml#L45-L54 But that's probably just a detail and we will align the permissions with this ADR as part of this story: [ASC-300](https://issues.redhat.com/browse/ASC-300) cc @ranakan19
2. if `deploymenttargetclasses` is a cluster-scoped resource, then we cannot use the template to grant the permissions - the template is applied in a namespace-scope only. In that case, the permissions for `deploymenttargetclasses` has to be granted to "all authenticated users" via ClusterRole and ClusterRoleBinding. @hongweiliu17 could you please remove `deploymenttargetclasses` from the template and create the ClusterRole(Binding) either on your side or via infra-deployments?

Reply:
Thanks for the details, it is very clear.

Reply:
I add DeployementTargetClass with addition info (`Cluster-scoped`)as below:
```
| | DeploymentTargetClass (Cluster-scoped) | appstudio.redhat.com | get, list, watch | deploymenttargetclasses
```